src/main/java/core/basesyntax/service/Injector.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.BetDaoImpl;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.dao.UserDaoImpl;
+import core.basesyntax.exceptions.NoAnnotationPresentException;
+import core.basesyntax.factory.Factory;
+import core.basesyntax.lib.Dao;
+import core.basesyntax.lib.Inject;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+    public static final String EXC_MESSAGE = "Annotation @Dao is absent in ";
+
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException, InstantiationException {
+        Field[] declaredFields = clazz.getDeclaredFields();
+        Object instance = null;
+        instance = clazz.getDeclaredConstructor().newInstance();
+
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                field.setAccessible(true);
+                if (field.getType().equals(BetDao.class)) {
+                    if (BetDaoImpl.class.isAnnotationPresent(Dao.class)) {
```

Comment:
This check can be united with the UserDaoImpl check and executed from the for loop because it has nothing in common with the fields. Agree?
It will help us to get rid of redundant code.